smt/surrogate_models/kpls.py

```diff
@@ -102,31 +104,33 @@ def _estimate_number_of_components(self):
                 # as the problem is the same and the data are mainly similar
                 if fold == 0:
                     super(KPLS, self)._initialize()
+                    self.options["corr"] = corr
                 else:
                     super(KPLS, self)._initialize()
+                    self.options["corr"] = corr
```

Comment:
Looking rapidly, not sure to understand why `self.options["corr"]` is assigned but the assignment could be factorized out after the if and the `super(KPLS, self)._initialize()` factorized out before.